File: RQ1/GitHub Copilot Chat/Python/WT3&T4 Case.py
# Source Code
def named_query(order_by=None, **conds):
    """
    Create a named query.
    """
    cond_str = "WHERE "
    for cond in conds.iterkeys():
        if len(cond_str) > 6:
            cond_str += ' AND '
        cond_str += '%s %s' % (cond, conds[cond])
    
    if order_by:
        cond_str += ' ORDER BY %s' % order_by
    
    return property(lambda self: cls.gql(cond_str))

# Synthesized Code, syntax similarity:0.20
def named_query(order_by=None, **conds):
    query = "SELECT * FROM table_name"
    if conds:
        query += " WHERE " + " AND ".join(f"{k} = '{v}'" for k, v in conds.items())
    if order_by:
        query += f" ORDER BY {order_by}"
    return query

File: RQ1/GitHub Copilot Chat/Java/WT3&T4 Case.java
/** Even indirectly */
//Source Code
public boolean isDerivedFrom(ClassInfo cl){
    ClassInfo dad = this.superclass();
    if (dad != null) {
        if (dad.equals(cl)) {
            return true;
        } else {
            if (dad.isDerivedFrom(cl)) {
                return true;
            }
        }
    }
    for (ClassInfo iface: interfaces()) {
        if (iface.equals(cl)) {
            return true;
        } else {
            if (iface.isDerivedFrom(cl)) {
                return true;
            }
        }
    }
    return false;
}

// Synthesized Code, syntax similarity: 0.18
public boolean isDerivedFrom(ClassInfo cl) {
    ClassInfo current = this;
    while (current != null) {
        if (current == cl) {
            return true;
        }
        current = current.getSuperClass();
    }
    return false;
}